internal/wrappers/features/step_defs_cpp/collection-steps.cpp

```diff
@@ -0,0 +1,209 @@
+#include "steps.hpp"
+
+// Anonymous namespace to handle a linker error
+// see https://stackoverflow.com/questions/14320148/linker-error-on-cucumber-cpp-when-dealing-with-multiple-feature-files
+namespace {
+  WHEN("^I create a collection \'([^\"]*)\'$")
+  {
+    REGEX_PARAM(std::string, collection_id);
+
+    ScenarioScope<KuzzleCtx> ctx;
+
+    try {
+      ctx->kuzzle->collection->create(ctx->index, collection_id);
+    } catch (KuzzleException e) {
+      BOOST_FAIL(e.getMessage());
+    }
+  }
+
+  THEN("^the collection \'([^\"]*)\' should exists$")
+  {
+    REGEX_PARAM(std::string, collection_id);
+
+    ScenarioScope<KuzzleCtx> ctx;
+
+    BOOST_CHECK(ctx->kuzzle->collection->exists(ctx->index, collection_id) == true);
+  }
+
+  WHEN("^I check if the collection \'([^\"]*)\' exists$")
+  {
+    REGEX_PARAM(std::string, collection_id);
+
+    ScenarioScope<KuzzleCtx> ctx;
+
+    ctx->success = ctx->kuzzle->collection->exists(ctx->index, collection_id);
+  }
+
+  THEN("^the collection should exist$")
+  {
+    ScenarioScope<KuzzleCtx> ctx;
+
+    BOOST_CHECK(ctx->success == true);
+
+    ctx->success = false;
+  }
+
+  GIVEN("^it has a collection \'([^\"]*)\'$")
+  {
+    REGEX_PARAM(std::string, collection_name);
+
+    ScenarioScope<KuzzleCtx> ctx;
+    ctx->collection = collection_name;
+
+    K_LOG_D("Creating collection: %s", collection_name.c_str());
+    try {
+      ctx->kuzzle->collection->create(ctx->index, ctx->collection);
+    } catch (KuzzleException e) {
+      K_LOG_E(e.getMessage().c_str());
+      BOOST_FAIL(e.getMessage());
+    }
+  }
+
+  WHEN("^I list the collections of \'([^\"]*)\'$")
+  {
+    REGEX_PARAM(std::string, index_name);
+
+    ScenarioScope<KuzzleCtx> ctx;
+
+    ctx->content = ctx->kuzzle->collection->list(index_name);
+
+    json_spirit::Value collectionsList;
+    json_spirit::read(ctx->content, collectionsList);
+    json_spirit::Value field_value = json_spirit::find_value(collectionsList.get_obj(), "collections");
+
+    ctx->hits = field_value.get_array().size();
+  }
+
+  GIVEN("^the collection has a document with id \'([^\"]*)\'$")
+  {
+    REGEX_PARAM(std::string, document_id);
+
+    ScenarioScope<KuzzleCtx> ctx;
+
+    query_options options = {0};
+    options.refresh = (char*)"wait_for";
```

Comment:
Why do we need a cast, and can you use a cast operator instead? 
Is it because `"wait_for"` is a `const char*`?

Reply:
C++ forbid conversion from string constant to `char*`.  
I use a `const_cast<char*>` instead now

Reply:
I was afraid of that answer.
This means either that
* there is a mistake, and `options.refresh` is of type `char*` instead of `const char *`. This should be fixed because we do not (and obviously should not) modify the provided option value
or:
* `options.refresh` is really of type `char*` and there is a `free` (or `delete`) in here somewhere. If that doesn't crash, it is either by sheer luck or because we forgot to free the value

Obviously I would prefer option 1.